src/Orleans.Runtime/MembershipService/GrainBasedMembershipTable.cs

```diff
@@ -31,7 +31,7 @@ public GrainBasedMembershipTable(IServiceProvider serviceProvider, ILogger<Grain
         {
             var options = this.serviceProvider.GetRequiredService<IOptions<DevelopmentMembershipOptions>>().Value;
             var siloDetails = this.serviceProvider.GetService<ILocalSiloDetails>();
-            bool isPrimarySilo = siloDetails.SiloAddress.Endpoint.Equals(options.PrimarySiloEndpoint);
+            bool isPrimarySilo = siloDetails.SiloAddress.Endpoint.Equals(options.PrimarySiloEndpoint) || options.PrimarySiloEndpoint == null;
```

Comment:
I think it should throw an exception if `PrimarySiloEndpoint` isn't set, maybe in a validator.

The reason is that if someone do not set any value for it and start two silos, it will be hard for this person to understand what's going on: the two silos will not be in the same cluster.

Reply:
should add a configuration validator to detect that earlier

Reply:
Hmm, the alternative is that when a use calls `UseLocalhostClustering` it automatically sets the primary silo endpoint to the current silo endpoint (equivalent to `ClusterConfiguration.LocalhostPrimarySilo()`), which doesn't really buy us anything.